src/sentry/tasks/assemble.py

```diff
@@ -353,64 +446,26 @@ def assemble_artifacts(org_id, version, checksum, chunks, project_ids=None, **kw
+            #  handling class.
             archive = ReleaseArchive(temp_file)
         except Exception:
             raise AssembleArtifactsError("failed to open release manifest")
 
         with archive:
             manifest = archive.manifest
 
-            org_slug = manifest.get("org")
-            if organization.slug != org_slug:
-                raise AssembleArtifactsError("organization does not match uploaded bundle")
-
-            release_name = manifest.get("release")
-            if version and release_name != version:
-                raise AssembleArtifactsError("release does not match uploaded bundle")
-
-            release = None
-            if version:
-                try:
-                    release = Release.objects.get(
-                        organization_id=organization.id, version=release_name
-                    )
-                except Release.DoesNotExist:
-                    raise AssembleArtifactsError("release does not exist")
-
-            dist_name = manifest.get("dist")
-            dist = None
-            if release and dist_name:
-                dist = release.add_dist(dist_name)
-
-            artifact_count = len(manifest.get("files", {}))
-            min_artifact_count = options.get("processing.release-archive-min-files")
-            saved_as_archive = False
-
-            # If we receive a version, it means that we want to create a ReleaseFile, otherwise we will create
-            # an ArtifactBundle.
-            if version and artifact_count >= min_artifact_count:
-                if release is None:
-                    raise AssembleArtifactsError("release does not exist")
-
-                try:
-                    update_artifact_index(release, dist, bundle)
-                    saved_as_archive = True
-                except Exception as exc:
-                    logger.error("Unable to update artifact index", exc_info=exc)
-            elif version is None:
-                _create_artifact_bundle(release, dist, org_id, project_ids, bundle, artifact_count)
-                saved_as_archive = True
-
-            if not saved_as_archive:
-                meta = {
-                    "organization_id": organization.id,
-                    "release_id": release.id,
-                    "dist_id": dist.id if dist else dist,
-                }
-                _store_single_files(archive, meta, True)
-
-            # Count files extracted, to compare them to release files endpoint
-            metrics.incr("tasks.assemble.extracted_files", amount=artifact_count)
+            if upload_as_artifact_bundle:
+                artifact_count = handle_assemble_for_artifact_bundle(
+                    bundle, manifest, organization, version, project_ids, dist
+                )
+            else:
+                artifact_count = handle_assemble_for_release_file(
+                    bundle, archive, manifest, organization, version
```

Comment:
```suggestion
                    bundle, manifest, organization, version, archive
```

Maybe we can keep it consistent like this?